nautobot/dcim/factory.py

```diff
@@ -0,0 +1,98 @@
+import logging
+
+import factory
+
+from nautobot.core.factory import OrganizationalModelFactory, PrimaryModelFactory
+from nautobot.dcim.models import DeviceType, Manufacturer, Platform
+from nautobot.dcim.choices import SubdeviceRoleChoices
+from nautobot.utilities.factory import random_instance, UniqueFaker
+
+
+logger = logging.getLogger(__name__)
+
+
+class ManufacturerFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Manufacturer
+        exclude = ("has_description",)
+
+    class Params:
+        unique_name = UniqueFaker("word")
+
+    # Slug isn't defined here since it will always inherit from name.
+    name = factory.LazyAttribute(lambda o: o.unique_name.title())
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+
+class ManufacturerGetOrCreateFactory(ManufacturerFactory):
+    class Meta:
+        django_get_or_create = ("name",)
+
+
+class PlatformFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Platform
+        exclude = ("has_manufacturer", "has_description", "has_napalm_driver", "has_napalm_args")
+
+    class Params:
+        first = UniqueFaker("word")
+        last = UniqueFaker("word")
+
+    # Slug isn't defined here since it will always inherit from name.
+    name = factory.LazyAttribute(lambda o: f"{o.first} {o.last}".title())
+
+    has_manufacturer = factory.Faker("pybool")
+    manufacturer = factory.Maybe("has_manufacturer", random_instance(Manufacturer), None)
+
+    has_napalm_driver = factory.Faker("pybool")
+    napalm_driver = factory.Maybe("has_napalm_driver", factory.Faker("locale"), "")
```

Comment:
Setting `napalm_driver` to real values might be a good reason to go ahead and use a static list of manufacturers. The list of napalm drivers is short and this might be easy to implement with [traits](https://factoryboy.readthedocs.io/en/stable/reference.html#traits)?


Napalm drivers:
- arista
  - eos
- juniper 
  - junos
- cisco
  - iosxr_netconf
  - iosxr
  - nxos
  - nxos_ssh
  - ios
- ideas for manufacturers with blank napalm_driver
  - brocade
  - dell
  - hp
  - aruba

Reply:
I'm not sure I'm clear on how traits would be useful here. I think there would still need to be the hard-coded list of manufacturers for `ManufacturerFactory`. Followed by a mapping of `manufacture`r => `napalm_driver` for use with `PlatformFactory` to randomly choose based on the manufacturer?